src/SprykerEco/Zed/Payone/Business/Payment/PaymentManager.php

```diff
@@ -1594,4 +1730,40 @@ protected function getPostSaveHookRequestContainer(
 
         return $paymentMethodMapper->mapPaymentToAuthorization($paymentEntity, $orderTransfer);
     }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayoneKlarnaStartSessionRequestTransfer $klarnaStartSessionRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\PayoneKlarnaSessionResponseTransfer
+     */
+    public function startKlarnaSession(PayoneKlarnaStartSessionRequestTransfer $klarnaStartSessionRequestTransfer): PayoneKlarnaSessionResponseTransfer
```

Comment:
* This class violates SOLID principles and is not maintainable.
* Such classes require refactoring.
* Let's create a new class `KlarnaPaymentManager` to handle klarna session.
* `KlarnaPaymentManager` extends `PaymentManager` class.

Reply:
Need discussion

Reply:
++